chain-registry/src/relayer_config.rs

```diff
@@ -0,0 +1,243 @@
+    let name_string = chain_name.to_string();
+    let assets_handle = tokio::spawn(async move { AssetList::fetch(name_string).await });
+
+    let chain_data = chain_data_handle.await.unwrap()?;
+
+    let grpc_endpoints: Vec<String> = chain_data
+        .apis
+        .grpc
+        .iter()
+        .map(|grpc| grpc.address.to_owned())
+        .collect();
+    let rpc_endpoints: Vec<String> = chain_data
+        .apis
+        .rpc
+        .iter()
+        .map(|rpc| rpc.address.to_owned())
+        .collect();
+
+    let rpc_handle = tokio::spawn(async move {
+        select_healthy(&rpc_endpoints, query_rpc, RegistryError::no_healthy_rpc()).await
+    });
+
+    let grpc_handle = tokio::spawn(async move {
+        select_healthy(
+            &grpc_endpoints,
+            health_check_grpc,
+            RegistryError::no_healthy_grpc(),
+        )
+        .await
+    });
+
+    let base = if let Some(asset) = assets_handle.await.unwrap()?.assets.first() {
+        asset.base.clone()
+    } else {
+        return Err(RegistryError::no_asset_found(chain_name.to_string()));
+    };
+
+    let rpc_mandatory_data = rpc_handle.await.unwrap()?;
+    let grpc_address = grpc_handle.await.unwrap()?;
+
+    Ok(ChainConfig {
+        id: ChainId::from_string(&chain_data.chain_name),
+        r#type: default::chain_type(),
+        rpc_addr: tendermint_rpc::Url::from_str(rpc_mandatory_data.rpc_address.as_str()).unwrap(),
+        websocket_addr: websocket_from_rpc(&rpc_mandatory_data.rpc_address)?,
+        grpc_addr: grpc_address,
+        rpc_timeout: default::rpc_timeout(),
+        account_prefix: chain_data.bech32_prefix,
+        key_name: key_name.to_string(),
+        key_store_type: Store::default(),
+        store_prefix: "ibc".to_string(),
+        default_gas: Some(100000),
+        max_gas: Some(400000),
+        gas_adjustment: None,
+        gas_multiplier: Some(1.1),
+        fee_granter: None,
+        max_msg_num: MaxMsgNum::default(),
+        max_tx_size: MaxTxSize::default(),
+        clock_drift: default::clock_drift(),
+        max_block_time: default::max_block_time(),
+        trusting_period: None,
+        memo_prefix: Memo::default(),
+        proof_specs: Default::default(),
+        trust_threshold: TrustThreshold::default(),
+        gas_price: GasPrice {
+            price: 0.1,
+            denom: base,
+        },
+        packet_filter: PacketFilter::default(),
+        address_type: AddressType::default(),
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::utils::TEST_CHAINS;
+
+    async fn fetch_chain(chain: &str) {
+        hermes_config(chain, "testkey").await.unwrap();
+    }
+
+    #[test]
+    fn test_fetch_chain_config() {
+        use tokio::runtime::Runtime;
+        let rt = Runtime::new().unwrap();
+
+        let mut handles = Vec::with_capacity(TEST_CHAINS.len());
+        for chain in TEST_CHAINS {
+            handles.push(fetch_chain(chain));
```

Comment:
This test confuses me. `hermes_config` returns a `Result<ChainConfig>`, but the `fetch_chain` function, according to its signature, doesn't return anything. But then here the output of `fetch_chain` is pushed to a vector. Upon checking the length of the vector when running it the test, it does contain elements inside, which is the confusing part. 

Changing the the signature of `fetch_chain` to specify that it returns a `ChainConfig` still works, and is less confusing, though it doesn't answer the question of why this works as-is in the first place.

Reply:
I have rewritten this test without using a helper function. I forgot to use `tokio::spawn()`. The vector should contain a `JoinHandle`.